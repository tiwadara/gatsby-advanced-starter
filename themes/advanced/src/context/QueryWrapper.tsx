import React from "react";

import { QueryClient, QueryClientProvider } from "react-query";
import { ReactQueryDevtools } from "react-query/devtools";

const queryClient = new QueryClient();

// Wrapper for react-query context that will be added to the app (both for SSR and Client)
const QueryWrapper = ({
  element,
}: {
  element: React.ReactChild;
}): JSX.Element => (
  <QueryClientProvider client={queryClient}>
    {element}
    <ReactQueryDevtools initialIsOpen={false} />
  </QueryClientProvider>
);

export default QueryWrapper;
