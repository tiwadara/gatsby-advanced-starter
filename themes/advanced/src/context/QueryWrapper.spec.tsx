import QueryWrapper from "./QueryWrapper";

describe("context QueryWrapper", () => {
  it("renders correctly", () => {
    expect.assertions(1);

    const ret = QueryWrapper({ element: "Test" });

    expect(ret).toMatchSnapshot();
  });
});
