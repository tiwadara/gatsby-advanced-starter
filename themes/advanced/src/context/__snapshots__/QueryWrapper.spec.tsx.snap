// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context QueryWrapper renders correctly 1`] = `
<QueryClientProvider
  client={
    QueryClient {
      "defaultOptions": Object {},
      "mutationCache": MutationCache {
        "config": Object {},
        "listeners": Array [],
        "mutationId": 0,
        "mutations": Array [],
      },
      "mutationDefaults": Array [],
      "queryCache": QueryCache {
        "config": Object {},
        "listeners": Array [],
        "queries": Array [],
        "queriesMap": Object {},
      },
      "queryDefaults": Array [],
    }
  }
>
  Test
  <ReactQueryDevtools
    initialIsOpen={false}
  />
</QueryClientProvider>
`;
